src/main/java/me/xemor/superheroes/skills/skilldata/DamageModifierData.java

```diff
@@ -69,34 +61,27 @@ public final boolean isValidCause(EntityDamageEvent.DamageCause damageCause) {
         // blacklist nonempty and it is NOT in it
         // whitelist nonempty and it is IN it
         // both are empty
-        return (!causeblacklist.getSet().isEmpty() && !causeblacklist.inSet(damageCause)) ||
-            (!causewhitelist.getSet().isEmpty() && causewhitelist.inSet(damageCause)) ||
-            (causewhitelist.getSet().isEmpty() && causeblacklist.getSet()
-                .isEmpty());
+        return (whitelist && !causes.inSet(damageCause));
     }
 
     public final boolean isValidEntity(EntityType entityType) {
         // blacklist nonempty and it is NOT in it
         // whitelist nonempty and it is IN it
         // both are empty
-        return (!entityblacklist.getSet().isEmpty() && !entityblacklist.inSet(entityType)) ||
-            (!entitywhitelist.getSet().isEmpty() && entitywhitelist.inSet(entityType)) ||
-            (entitywhitelist.getSet().isEmpty() && entityblacklist.getSet()
-                .isEmpty());
+        return (whitelist && !entities.inSet(entityType));
     }
 
-    public final double calculateDamage(double damage) {
-        double x = damage;
+    public final double calculateDamage(double x) {
```

Comment:
why did you change the variable name to x, that's what I meant by the obfuscation. Use a sensible name